example/fwup-architecture/image.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <libopencm3/cm3/vector.h>
+#include <stdint.h>
+
+#define IMAGE_MAGIC 0xb1accafe
+
+typedef enum {
+    IMAGE_TYPE_LOADER = 0x1,
```

Comment:
what's the distinction between a LOADER & UPDATER?

Reply:
There are four binary types:
1. Boot: the primary bootloader. Not updatable hence not in this enum. All it does is load the Loader.
2. Loader: the secondary bootloader. In normal operation it loads the app, it also knows how to update it
3. App: the main application
4. Updater: A special application which nows how to update the Loader.